### PR TITLE
Enhance .gitignore

### PR DESCRIPTION
**Pull request by jooola**
Removed some duplicate entries, and split the root gitignore in multiple ones.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,163 @@
...
+
+# Flask stuff:
+instance/
+.webassets-cache
+
+# Scrapy stuff:
+.scrapy
+
+# Sphinx documentation
+docs/_build/
```
Comment (paddatrapper) [inline .gitignore:97]: Should this be moved to `docs/.gitignore`?

File: .gitignore [hunk 2]
```diff
@@ -1,23 +1,163 @@
...
+
+# Environments
+.env
+.venv
+env/
+venv/
+ENV/
+env.bak/
+venv.bak/
+
```
Comment (paddatrapper) [inline .gitignore:140]: While we're listing python virtualenvs, I'd add pyenv to this list

File: .gitignore (hunk 1 above)
Comment (jooola) [reply .gitignore:97]: I copy pasted this whole section from the Github file, so I didn't want to touch it. I would leave it as is, but I am ok with moving it in the docs/.gitignore

File: .gitignore (hunk 2 above)
Comment (jooola) [reply .gitignore:141]: Done

File: .gitignore (hunk 1 above)
Comment (jooola) [reply .gitignore]: I commented it out.

Comment (paddatrapper) [reply .gitignore]: I think you can remove it, just needs to be in the docs one

Comment (jooola) [reply .gitignore]: But we don't use sphinx documentation right ?

Comment (paddatrapper) [reply .gitignore]: Good point, not anymore. So actually can be removed completely

**Pull request merged**